consensus/istanbul/core/roundstate_db.go

```diff
@@ -172,5 +226,70 @@ func (rsdb *roundStateDBImpl) GetRoundStateFor(view *istanbul.View) (RoundState,
 }
 
 func (rsdb *roundStateDBImpl) Close() error {
+	if rsdb.opts.withGarbageCollector {
+		rsdb.stopGarbageCollector()
+	}
 	return rsdb.db.Close()
 }
+
+func (rsdb *roundStateDBImpl) garbageCollectEntries() {
+	logger := rsdb.logger.New("func", "StartGarbageCollector")
+
+	oldestValidView, err := rsdb.GetOldestValidView()
+	if err != nil {
+		logger.Error("Aborting RoundStateDB GarbageCollect: Failed to fetch oldestValidView", "err", err)
+		return
+	}
+
+	logger.Debug("Prunning entries from old views", "oldestValidView", oldestValidView)
+	count, err := rsdb.deleteEntriesOlderThan(oldestValidView)
+	if err != nil {
+		logger.Error("Aborting RoundStateDB GarbageCollect: Failed to remove entries", "entries_removed", count, "err", err)
+		return
+	}
+
+	logger.Info("Finished RoundStateDB GarbageCollect", "removed_entries", count)
+}
+
+func (rsdb *roundStateDBImpl) deleteEntriesOlderThan(lastView *istanbul.View) (int, error) {
+	fromViewKey := view2Key(&istanbul.View{Sequence: common.Big0, Round: common.Big0})
+	toViewKey := view2Key(lastView)
+
+	iter := rsdb.db.NewIterator(&util.Range{Start: fromViewKey, Limit: toViewKey}, nil)
```

Comment:
Very cool that leveldb can do this

Reply:
Yes! key are sorted, and you can define the comparator... but soring happens at byte level, so i had to think about sort order when the key is serialized